src/shims/unix/fs.rs

```diff
@@ -366,6 +366,18 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 return Ok(-1);
             }
         }
+
+        #[cfg(unix)]
```

Comment:
This means that Miri running on Linux will have different support for `x86_64-unknown-linux-gnu` than Miri running on Windows. I do not think we want that. Can we support this on all host platforms?

Reply:
I am not very familiar with Windows, so I am not sure how to properly emulate it and whether it's possible in the first place. Maybe we could simply do nothing on Windows for this flag?

Reply:
You don't need to know Windows, all you need to do is figure out how to implement this flag with portable standard library APIs, without passing through to the host. Based on the documentation of `O_NOFOLLOW` this seems quite possible.

Reply:
You mean by adding a default impl which uses [`Path::is_symlink`](https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.is_symlink)?

Reply:
I decided to add a warning that `O_NOFOLLOW` is ignored on non-UNIX targets. There is already a bunch of code gated on `unix` in this module, so it's not like this is a sole exception. A `Path`-based emulation can be added in a later PR, I can create an issue for it later.